packages/assets-controllers/src/TokenDetectionController.ts

```diff
@@ -189,7 +269,26 @@ export class TokenDetectionController extends StaticIntervalPollingController<
           useTokenDetection &&
           (isSelectedAddressChanged || isDetectionChangedFromPreferences)
         ) {
-          await this.detectTokens();
+          await this.#restartTokenDetection({
+            selectedAddress: this.#selectedAddress,
+          });
+        }
+      },
+    );
+
+    this.messagingSystem.subscribe(
```

Comment:
Ah. Hopefully once the PreferencesController -> AccountsController migration is complete for selected address we don't have to check for the selected address in the PreferencesController listener above. But this is good to subscribe to this in addition to PreferencesController in the meantime.